### ISSUE TITLE
Core validator 'compare' always return false (!= issue 9633)

### ISSUE THREAD

**Issue by MigueSchpeir** [yiisoft/yii2#10230]
As @cebe sugested me, I'm opening a new issue with my problem.

My model validators:
```
[['quantity_start', 'quantity_end'], 'integer'],
[['quantity_end'], 'compare', 'compareAttribute' => 'quantity_start', 'operator' => '<=', 'type' => 'number'],
```

But, as you can see in the following picture, the validator always returns false (with greater, smaller or equal values):

![](https://cloud.githubusercontent.com/assets/7605756/11300706/e47c5036-8f6f-11e5-869e-92679e796dfd.jpg)

I don't know if it's related but i'm currently using [this wiki](http://www.yiiframework.com/wiki/666/handling-tabular-data-loading-and-validation-in-yii-2) to persist model data.

**Comment by SilverFire** [comment]
Failed to reproduce with basic Yii application template.

We need more info. Please, show your action and view code.

**Comment by SilverFire** [comment]
Closed since topic stared is inactive.

**Comment by MigueSchpeir** [comment]
Sorry for the delay...

## Action
```
/**
     * @param integer $id
     * @return mixed
     */
    public function actionOpen($id)
    {
        ...
        try {
           ...
            if (Model::loadMultiple($items, Yii::$app->request->post())) {
                ...                
                $count = 0;
                foreach ($items as $item) {                    
                    $item->id_bar = $id;
                    $item->date_start = $date;
                    if ($item->save()) {
                        $count++;
                    } else {
                        throw new \Exception('...');
                    }
                }
                Yii::$app->session->setFlash('success', "...");
                return $this->redirect(['/bar/view', 'id' => $id]);
            }
        } catch (\Exception $exc) {
            Yii::$app->getSession()->setFlash('danger', $exc->getMessage());            
            return $this->redirect(['/bar/view', 'id' => $id]);
        }

        return $this->render('open',
                [
                'bar' => $bar,
                'items' => $items,
        ]);
    }
```

## View

```
<table class="table table-striped table-bordered" >
            <colgroup>
                <col style="width: 60%">
                <col style="width: 20%">
                <col style="width: 20%">
            </colgroup>
            <tr>
                <th>Producto</th>
                <th>Cantidad Inicial</th>
                <th>Cantidad Final</th>
            </tr>
            <?php foreach($items as $i=>$item): ?>
                <tr>
                    <td><strong><?= Product::findOne($item->id_product)->getFullName()  ?></strong></td>
                    <td><?= $form->field($item,"[$i]quantity_start")->textInput($mobileNumberOptions)->label(false); ?></td>
                    <td><?= $form->field($item,"[$i]quantity_end")->textInput($mobileNumberOptions)->label(false); ?></td>
                </tr>
            <?php endforeach; ?>
        </table>
```

**Comment by SilverFire** [comment]
Do you use JS or Ajax validation?

**Comment by MigueSchpeir** [comment]
I think Ajax validation is set by default with

```php
$form = ActiveForm::begin(); 
```

right?

**Comment by SilverFire** [comment]
No, it's disabled by default: https://github.com/yiisoft/yii2/blob/master/framework/widgets/ActiveForm.php#L106

**Comment by MigueSchpeir** [comment]
@SilverFire well, JS then. This is the olny view that's giving me that kind of problem.

**Comment by SilverFire** [comment]
I'm sorry, but it's really hard to guess, what might be a problem.
Could you try to reproduce the bug on clean Yii2 application, because I'm pretty sure there is some mistake in your code, that interacts on this page.

**Comment by MigueSchpeir** [comment]
@SilverFire, I've tried with a clean basic app. I'm having the same issue.

Another thing I've tried was change this (multiple model):

```php
<?php foreach($items as $i=>$item): ?>
      <tr>
            <td><strong><?= Product::findOne($item->id_product)->getFullName()  ?></strong></td>
            <td><?= $form->field($item,"[$i]quantity_start")->textInput($mobileNumberOptions)->label(false); ?></td>
            <td><?= $form->field($item,"[$i]quantity_end")->textInput($mobileNumberOptions)->label(false); ?></td>
      </tr>
<?php endforeach; ?>
```

for this (single model):
```php
<?= Product::find()->one()->getFullName()  ?>
<?= $form->field($model,"[]quantity_start")->textInput($mobileNumberOptions)->label(false); ?>
<?= $form->field($model,"[]quantity_end")->textInput($mobileNumberOptions)->label(false); ?>
```

and the validation worked!

I think it is something related to that ``` [$i] ```.

I'll wait for your suggestions.

**Comment by SilverFire** [comment]
Well, it duplicates #9811. The funniest thing is that I reported this bug before I joined core time and forgot about it now :smile:

**Comment by SilverFire** [comment]
I'm closing this bug as a duplicate, but feel free to continue discussion in #9811.

Thank you for your report :)

**Comment by karelvasicek** [comment]
@samdark , I found this issue as I had faced the same problem - not working client validation for tabular form. This issue was closed as a duplicate of #9811 and #9811 was later transferred to `yiisoft/validator` package and then closed too :). Was it intended?

**Comment by samdark** [comment]
Likely it was not :)
